x-pack/plugins/reporting/server/lib/esqueue/helpers/create_client.js

```diff
@@ -1,24 +0,0 @@
-/*
```

Comment:
Hey @tsullivan, as we agreed on Slack, here is the quick summary of what I'd like to do ES plugin in general and reporting plugin specifically:

We're going to replace ES plugin with the core ES service provided by the new platform. For some time these will coexist for BWC reasons. But legacy plugin will use ES service under the hood anyway, so we're trying to limit the API exposed by the legacy plugin and expose only what we need and gradually enhance/expand API in a more controlled manner. That means we'd like to avoid exposing "raw" Elasticsearch JS client to consumers and hence remove `Cluster.getClient` method used by Reporting (`Esqueue`) currently (and Monitoring as well, but the usage there was more contained and easier to update).

Removing `Cluster.getClient` usage from `Esqueue` is no-brainer, but the problem is that `Esqueue` can [create JS client](https://github.com/elastic/kibana/blob/6084acc8b6f0c427171f5074389654eb97a95149/x-pack/plugins/reporting/server/lib/esqueue/index.js#L30) if it doesn't receive it through arguments implying that the `client` from options should be of the same type/interface as "raw" Elasticsearch JS client. I haven't seen any code in Kibana that wouldn't pass `client` via options, but there is a chance I'm missing something. 

So my question is: is there any reason why we'd want to support `options.client || createClient()` in `Esqueue`? If not, please see the reporting plugin related changes in this PR that I'd like to make and let me know if you have any concerns.

Reply:
Hi @azasypkin I also am not seeing and code in Kibana that wouldn't pass a `client` option to the Esqueue constructor, and unfortunately I can't answer why the constructor has that conditional assignment. I think the history of how the code got to be that way goes back to before Esqueue was pulled in from being an external dependency

Reply:
See: https://github.com/elastic/esqueue/blob/deprecated/src/index.js#L24

Reply:
Thanks @tsullivan! Awesome, I'll go ahead and remove `createClient` then.